### PR TITLE
feature/174379329-update-tsne-dao-to-contact-latest-schema

### PR DESCRIPTION
**Pull request by upendrakumbham**
Update SELECT queries as per the latest schema and test data. This PR is under testing. This PR is a companion PR of the SCXA PR - [Update tne-dao](https://github.com/ebi-gene-expression-group/atlas-web-single-cell/pull/115)

### CODE REVIEW

File: src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java [hunk 1]
```diff
@@ -15,190 +14,208 @@
...
+		return jdbcTemplate.queryForList(
+				"SELECT cell_id FROM scxa_analytics  WHERE experiment_accession=? ORDER BY RANDOM() LIMIT ?",
+				String.class,
+				experimentAccession,
+				numberOfCells);
+	}
+
+	public int fetchRandomPerplexityFromExperimentTSne(String experimentAccession) {
+		return jdbcTemplate.queryForObject(
+				"SELECT perplexity FROM scxa_tsne WHERE experiment_accession=? ORDER BY RANDOM() LIMIT 1",
```
Comment (pcm32) [inline src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java:170]: This query points to older tables, can you check if this method is still used elsewhere and deprecate or remove (I don't know what is your current policy). I'm guessing there is somewhere a method with the newer table that replaces this.

File: src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java [hunk 2]
```diff
@@ -15,190 +14,208 @@
...
+				Integer.class,
+				experimentAccession);
+	}
+
+	public int fetchRandomPerplexityFromExperimentTSne(String experimentAccession, String geneId) {
+		return jdbcTemplate.queryForObject(
+				"SELECT perplexity FROM scxa_tsne AS tsne " +
```
Comment (pcm32) [inline src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java:177]: Also pointing to older scxa_tsne table.

File: src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java [hunk 3]
```diff
@@ -15,190 +14,208 @@
...
+						"LEFT JOIN scxa_analytics AS analytics " +
+						"ON analytics.experiment_accession=tsne.experiment_accession AND analytics.cell_id=tsne.cell_id " +
+						"WHERE tsne.experiment_accession=? AND analytics.gene_id=? ORDER BY RANDOM() LIMIT 1",
+				Integer.class,
+				experimentAccession, geneId);
+	}
+
+	public int fetchRandomKFromCellClusters(String experimentAccession) {
+		return jdbcTemplate.queryForObject(
+				"SELECT k FROM scxa_cell_clusters WHERE experiment_accession=? ORDER BY RANDOM() LIMIT 1",
```
Comment (pcm32) [inline src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java:187]: Pointing to old clusters table.

File: src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java [hunk 4]
```diff
@@ -15,190 +14,208 @@
...
+				Integer.class,
+				experimentAccession);
+	}
+
+	public List<Integer> fetchKsFromCellClusters(String experimentAccession) {
+		return jdbcTemplate.queryForList(
+				"SELECT DISTINCT(k) FROM scxa_cell_clusters WHERE experiment_accession=?",
```
Comment (pcm32) [inline src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java:194]: Pointing to old clusters table.

**Review by pcm32 — COMMENTED** [review]
Looks good in general, but there are some methods remaining that are pointing to old tables, I guess that if these methods are not being used anymore because they are replaced by logic in the single cell module, then you might want to remove or deprecate as per your usual policy.

File: src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java (hunk 4 above)
Comment (upendrakumbham) [reply src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java:196]: added @deprecated annotation on a method

File: src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java (hunk 3 above)
Comment (upendrakumbham) [reply src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java:188]: added @deprecated annotation on a method

**Review by upendrakumbham — COMMENTED** [review]
Sorry, accidentally I have clicked on started a review.  I have to approve it myself :).

File: src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java (hunk 2 above)
Comment (upendrakumbham) [reply src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java:177]: I am ignoring this as per our discussion.

File: src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java (hunk 1 above)
Comment (upendrakumbham) [reply src/main/java/uk/ac/ebi/atlas/testutils/JdbcUtils.java:170]: I am ignoring this as per our discussion. I talk to Jon more on this.

**Pull request merged**